src/core/thread/network_data_service.hpp

```diff
@@ -0,0 +1,362 @@
+     * changes in local Network Data with leader.
+     *
+     * The template type `ServiceType` has the following requirements:
+     *   - It MUST have a constant `ServiceType::kServiceNumber` specifying the service number.
+     *
+     * @tparam   ServiceType       The service type to be removed.
+     *
+     * @retval OT_ERROR_NONE       Successfully removed the Service entry.
+     * @retval OT_ERROR_NOT_FOUND  Could not find the Service entry.
+     *
+     */
+    template <typename ServiceType> otError Remove(void) { return RemoveService(ServiceType::kServiceNumber); }
+
+#endif
+
+    /**
+     * This method gets the Service ID for the specified service from Thread Network Data.
+     *
+     * The template type `ServiceType` has the following requirements:
+     *   - It MUST have a constant `ServiceType::kServiceNumber` specifying the service number.
+     *
+     * @tparam     ServiceType     The service type to be added.
+     *
+     * @param[in]  aServerStable   The Stable flag value for Server TLV
+     * @param[out] aServiceId      A reference where to put the Service ID.
+     *
+     * @retval OT_ERROR_NONE       Successfully got the Service ID.
+     * @retval OT_ERROR_NOT_FOUND  The specified service was not found.
+     *
+     */
+    template <typename ServiceType> otError GetServiceId(bool aServerStable, uint8_t &aServiceId) const
+    {
+        return GetServiceId(ServiceType::kServiceNumber, aServerStable, aServiceId);
+    }
+
+#if (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
+    /**
+     * This method gets the Primary Backbone Router (PBBR) in the Thread Network Data.
+     *
+     * @param[out]  aConfig      The Primary Backbone Router configuration.
+     *
+     * @retval OT_ERROR_NONE       Successfully got the Primary Backbone Router configuration.
+     * @retval OT_ERROR_NOT_FOUND  No Backbone Router Service in the Thread Network.
+     *
+     */
+    otError GetBackboneRouterPrimary(ot::BackboneRouter::BackboneRouterConfig &aConfig) const;
+#endif
+
+    /**
+     * This method gets the next SRP server info from the Thread Network Data "SRP Server Service" entries.
+     *
+     * This method allows caller to iterate through all server entries for Network Data "SRP Server Service". To get
+     * the first entry @p aIterator should be cleared (e.g., a new instance of `Iterator` or calling `Clear()` method).
+     *
+     * @param[out]   aInfo         A reference to `SrpServer::Info` to return the next SRP server info.
+     * @param[inout] aIterator     A reference to an iterator.
+     *
+     * @retval OT_ERROR_NONE       Successfully got the next SRP server info. @p aInfo and @p aIterator are updated.
+     * @retval OT_ERROR_NOT_FOUND  No more SRP server entries in Network Data.
+     *
+     */
+    otError GetNextSrpServerInfo(SrpServer::Info &aInfo, Iterator &aIterator) const;
```

Comment:
nit: put `inout` parameters before `out`
```suggestion
    otError GetNextSrpServerInfo( Iterator &aIterator, SrpServer::Info &aInfo) const;
```